Documentation/git-checkout-index.txt

```diff
@@ -59,6 +61,10 @@ OPTIONS
 	write the content to temporary files.  The temporary name
 	associations will be written to stdout.
 
+--sparse::
```

Comment:
This feels a little odd. From what I understand, we generally use the `--sparse` flag to enable sparse behavior, but here we seem to be circumventing it. We may want to consider something more like `--ignore-sparse`. Let me know if I'm missing something, though!

Reply:
Check out gitgitgadget/git#1018 which restricts how Git operates outside of the sparse-checkout. The `--sparse` option allows overriding these protections. It's _really new_.

Reply:
@ldennington I definitely see where you're coming from (it does seem somewhat backwards that `--sparse` would expand the index/do things outside the sparse checkout). I did base it on @derrickstolee's linked PR (+`ls-trees`), and I think this line conveys it best:

> a new '--sparse' option is added that ignores the sparse-checkout patterns and the SKIP_WORKTREE bit

The option here is pretty much treating `--sparse` as "does things in parts of the repository that should be sparse", rather than ignoring those parts of the repository. @derrickstolee please correct me if I'm using it wrong, though - I'm mostly looking for this to be consistent with the other already-implemented options (and offer a way to use `checkout-index --all` without always expanding the full index).

Reply:
@vdye: I think you have the right approach. We should make Git ignore files outside of the sparse-checkout cone as much as possible, and allow users to override that behavior by specifying `--sparse`. We do not expect users to have a legitimate reason to care about files outside of the sparse-checkout, but Git needs to care unless we make these changes.